velox/common/fuzzer/ConstrainedGenerators.cpp

```diff
@@ -222,19 +226,149 @@ getRandomInputGenerator(size_t seed, const TypePtr& type, double nullRatio) {
+    case TypeKind::TINYINT:
+    case TypeKind::SMALLINT:
+    case TypeKind::INTEGER:
+    case TypeKind::BIGINT:
+    case TypeKind::REAL:
+    case TypeKind::DOUBLE:
+    case TypeKind::TIMESTAMP:
+      return;
+    case TypeKind::VARCHAR:
+    case TypeKind::VARBINARY:
+      if (makeRandomVariation_ && coinToss(rng_, 0.5)) {
+        path += ".length()";
+      }
+      return;
+    case TypeKind::ARRAY:
+      if (makeRandomVariation_ && coinToss(rng_, 0.1)) {
+        path += ".length()";
+      } else if (coinToss(rng_, 0.2)) {
+        path += fmt::format("[{}]", generateRandomIndex());
+        generateImpl(path, type->childAt(0));
+      } else if (makeRandomVariation_ && coinToss(rng_, 0.2)) {
+        path += "[]";
+        generateImpl(path, type->childAt(0));
+      } else if (makeRandomVariation_ && coinToss(rng_, 0.4)) {
+        // Generae slices.
+        auto emptySelector = "";
+        auto start = coinToss(rng_, 0.2)
+            ? emptySelector
+            : fmt::format("{}", generateRandomIndex());
+        auto end = coinToss(rng_, 0.2)
+            ? emptySelector
+            : fmt::format("{}", generateRandomIndex());
+        auto step = coinToss(rng_, 0.2)
+            ? emptySelector
+            : fmt::format(
+                  "{}",
+                  rand<int32_t>(
+                      rng_, -maxContainerLength_, maxContainerLength_));
+        path += fmt::format("[{}:{}:{}]", start, end, step);
+        generateImpl(path, type->childAt(0));
+      }
+      return;
+    case TypeKind::ROW: {
+      const auto selectedField =
+          rand<uint64_t>(rng_, 0, type->asRow().size() - 1);
+      if (makeRandomVariation_ && coinToss(rng_, 0.2)) {
+        path += ".length()";
+      } else if (coinToss(rng_, 0.3)) {
+        path += fmt::format("[{}]", selectedField);
+        generateImpl(path, type->childAt(selectedField));
+      } else if (makeRandomVariation_ && coinToss(rng_, 0.3)) {
+        path += "[]";
+        generateImpl(path, type->childAt(selectedField));
+      }
+      return;
+    }
+    case TypeKind::MAP: {
+      const auto selectedKey =
+          mapKeys_[rand<uint64_t>(rng_, 0, mapKeys_.size() - 1)].toString(
+              type->childAt(0));
+      if (makeRandomVariation_ && coinToss(rng_, 0.15)) {
+        path += ".length()";
+      } else if (makeRandomVariation_ && coinToss(rng_, 0.15)) {
+        path += fmt::format("[{}]", selectedKey);
+        generateImpl(path, type->childAt(1));
+      } else if (coinToss(rng_, 0.15)) {
+        path += fmt::format("['{}']", selectedKey);
+        generateImpl(path, type->childAt(1));
+      } else if (makeRandomVariation_ && coinToss(rng_, 0.15)) {
+        path += "[]";
+        generateImpl(path, type->childAt(1));
+      } else if (coinToss(rng_, 0.15)) {
+        path += "[*]";
+        generateImpl(path, type->childAt(1));
+      } else if (coinToss(rng_, 0.15)) {
+        path += fmt::format(".{}", selectedKey);
```

Comment:
as a follow up we can add the option of appending a key after brackets without a dot which is valid syntax but not currently supported in velox, eg,`$.[2]payload`

Reply:
@bikramSingh91 Could you give an example of what `$.[2]payload` means? Is it that the `$` is an array of object, so `[2]` selects the second object in the array, then `payload` selects a key inside this object?

Reply:
Yes, its equivalent to `$.[2].payload`